pilot/pkg/xds/xds_cache.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Istio Authors
```

Comment:
maybe move the cache to a separate package

Reply:
and make it more generic

Reply:
I don't know - attempting to make things too generic can result in both complexity and lost optimization opportunities. It's mostly boilerplate - no rocket science algo...